extensions/expanded-opcodes/README.md

```diff
@@ -11,77 +11,52 @@ This extension adds an expanded instruction format to allow for a larger number
-
-
-1) Enables NEG and NOT to be encoded as `RSUB r, imm`.
-3) Designed to allow for building a larger immediate value. To reach a full 16 bit immediate, a 4th `SLO` or an additional `NOT` may be required.
-4) Primary intent is that these are used with immediate. Exact assignment of control registers is still floating. At least the the CPU status/extension/feature control registers should be present.
-5) The C and O flags are in an undefined state after execution of these instructions. Implementations may do whatever is easiest. An extension may mandate a particular behavior, with good enough reason, but must *not* mandate that the value of these flags after the operation depends on their value before the operation.
-6) These instructions do not have a 2 register mode.
-7) This instruction zero extends argument B as if the value read is of the size specified by the SS bits in the instruction (assuming the relevant feature is present)
-8) This instruction sign extends argument B as if the value read is of the size specified by the SS bits in the instruction (assuming the relevant feature is present)
+
+### Opcode table
+
+| `C CCCC CCCC` | NAME            | Operation                          | Flags  | Comment     |
+|---------------|-----------------|------------------------------------|--------|-------------|
+| `0 0000 0000` | `ADC`           | `A ← A + B + C`                    | `ZNCV` | (2)         |
+| `0 0000 0001` | `SBB`           | `A ← A - B - C`                    | `ZNCV` | (2)         |
+| `0 0000 0010` | `RSBB`          | `A ← B - A - C`                    | `ZNCV` | (1) (2)     |
+| `0 0000 0011` |                 |                                    |        | reserved    |
+| `0 0000 01xx` |                 |                                    |        | reserved    |
+| `0 0000 1xxx` |                 |                                    |        | reserved    |
+| `0 0001 xxxx` |                 |                                    |        | reserved    |
+| `0 001x xxxx` |                 |                                    |        | reserved    |
+| `0 01xx xxxx` |                 |                                    |        | reserved    |
+| `0 1xxx xxxx` |                 |                                    |        | reserved    |
+| `1 xxxx xxxx` |                 |                                    |        | reserved    |
+
+
+1) Enables NEG and NOT to be encoded as `RSBB r, imm` for integers larger than supported by the ISA.
+2) C in the operation refers to the carry bit.
+
+## Jump and Call
+
+| First byte   | 2nd-9th Byte | Comment                                                             |
+|:-------------|:-------------|:--------------------------------------------------------------------|
+| `111 10 JJJ` | `DDDD DDDD`  | Relative jump                                                       |
+| `111 11 JJJ` | `DDDD DDDD`  | Relative call if stack extension is implemented, otherwise reserved |
```

Comment:
I really do not like this `JJJ` thing. It's a heavy decoding cost (not just figuring out how long the instruction is, but also needing to support _handling_ instructions of all those lengths) for very little practical benefit. We could instead have
```
111 ????? 1J ?? CCCC (rmi)|(DDDD)+
```
There's lots of things these `?` bits could be used for to get many different types of jumps and calls; I'd suggest simply forcing them to particular values and letting everything else be reserved. For example, they could make the transfers switch between absolute/relative, they could make the address/displacement come from an `rmi` byte instead of a literal displacement (i.e. from the AAA register or from a memory operand). I wouldn't want to allow displacements of sizes other than 1, 2, and 4 bytes, and absolute addresses should be the same size as the current pointer mode.

With this approach, the conditional prefix should not be allowed to appear on any of these, since they always have a condition code regardless.

Reply:
If you're really that against being able to use any 1-8 byte displacement, maybe making it a logarithmic thing so that it's 1, 2, 4, and 8 for 2 of the J bits and then the 3rd will select between relative and absolute. I don't see an issue with requiring the conditional prefix to make this conditional since it saves space.

Reply:
It's less about not requiring a prefix and more about putting the FF bits in a consistent place. Taking a cue from RISC-V, which goes to some seemingly strange lengths to put common subsets of instructions in the same place.

Reply:
This still follows that. The mode selection is in a different location for expanded opcodes though which would be required anyway.